Common/Classes/Binary/Binary.cs

```diff
@@ -1578,10 +1578,15 @@ public static UInt64 ReadUInt64LSB(byte[] data, int byteOffset, int bitCount, re
                 int byteCount = (byte)(totalBitCount >> Common.Binary.Tres); // totalBitCount / 8
                 if (totalBitCount % Common.Binary.BitsPerByte > Common.Binary.Zero) ++byteCount;
 
-                // Check if we won't read more bytes than there are available.
-                if (byteCount > (data.Length - byteOffset))
+                // Check if we won't read less or more bytes than there are available.
+                if (byteCount < 0 || byteCount > (data.Length - byteOffset))
```

Comment:
Can byteCount ever be less than 0?

Reply:
Yep, you are correct, it can't.

```
int byteCount = (byte)(totalBitCount >> Common.Binary.Tres);
```

It is always >= 0